DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                    FailedLoginCount,
+                    HasBeenPromptedForPrn,
+                    LearningHubAuthId,
+                    HasDismissedLhLoginWarning,
+                    EmailVerified,
+                    DetailsLastChecked
+                )
+                SELECT Email,
+                    Password_deprecated,
+                    Forename_deprecated,
+                    Surname_deprecated,
+                    10,
+                    NULL,
+                    ProfileImage_deprecated,
+                    Active,
+                    ResetPasswordID_deprecated,
+                    TCAgreed_deprecated,
+                    FailedLoginCount_deprecated,
+                    0,
+                    NULL,
+                    0,
+                    GETUTCDATE(),
+                    GETUTCDATE()
+                    FROM AdminAccounts"
+            );
+
+            // 3. Update AdminAccounts to reference Users.ID
+            connection.Execute(
+                @"UPDATE AdminAccounts
+                    SET
+                        UserID = (SELECT ID FROM Users WHERE Email = PrimaryEmail),
+                        Email = NULL"
+            );
+
+            // 4. DelegateAccount
+
+            // Transfer all delegates with unique emails not already in the Users table
+            connection.Execute(
+                @"INSERT INTO dbo.Users (
+                    PrimaryEmail,
+                    PasswordHash,
+                    FirstName,
+                    LastName,
+                    JobGroupID,
+                    ProfessionalRegistrationNumber,
+                    ProfileImage,
+                    Active,
+                    ResetPasswordID,
+                    TermsAgreed,
+                    FailedLoginCount,
+                    HasBeenPromptedForPrn,
+                    LearningHubAuthId,
+                    HasDismissedLhLoginWarning,
+                    EmailVerified,
+                    DetailsLastChecked
+                )
+                SELECT Email,
+                    COALESCE(OldPassword, ''),
+                    COALESCE(FirstName_deprecated, ''),
+                    COALESCE(LastName_deprecated, ''),
+                    JobGroupID_deprecated,
+                    ProfessionalRegistrationNumber_deprecated,
+                    ProfileImage_deprecated,
+                    Active,
+                    ResetPasswordID_deprecated,
+                    NULL,
+                    0,
+                    HasBeenPromptedForPrn_deprecated,
+                    LearningHubAuthID_deprecated,
+                    HasDismissedLhLoginWarning_deprecated,
+                    GETUTCDATE(),
+                    GETUTCDATE()
+                    FROM DelegateAccounts
+                    WHERE Email IN (
+                        SELECT Email FROM DelegateAccounts
+                        WHERE Email IS NOT NULL AND TRIM(Email) IS NOT NULL AND TRIM(Email) <> ''

```

Comment:
Is "Trim(Email) is not null" not enough?

Reply:
TRIM('   ') returns '', I figured this was the safest way to be absolutely sure I caught them all, but if anything I need to remove the TRIM(Email) IS NOT NULL

Reply:
Yeah fair enough, feel free to leave as is.